src/TSID/src/SE3Task.cpp

```diff
@@ -0,0 +1,145 @@
+        return false;
+    }
+
+    // set the gains for the controllers
+    double kpLinear, kdLinear;
+    double kpAngular, kdAngular;
+    if (!ptr->getParameter("kp_linear", kpLinear) || !ptr->getParameter("kd_linear", kdLinear))
+    {
+        std::cerr << errorPrefix << descriptionPrefix << frameName
+                  << " - Error while the gains for the position controller are retrieved."
+                  << std::endl;
+        return false;
+    }
+
+    if (!ptr->getParameter("kp_angular", kpAngular) || !ptr->getParameter("kd_angular", kdAngular))
+    {
+        std::cerr << errorPrefix << descriptionPrefix << frameName
+                  << " - Error while the gains for the rotation controller are retrieved."
+                  << std::endl;
+        return false;
+    }
+
+    m_R3Controller.setGains({kpLinear, kdLinear});
+    m_SO3Controller.setGains({kpAngular, kdAngular});
+
+    // set the description
+    m_description = std::string(descriptionPrefix) + frameName + ".";
+
+    // resize the matrices
+    m_A.resize(6, variablesHandler.getNumberOfVariables());
+    m_A.setZero();
+    m_b.resize(6);
+    m_jacobian.resize(6, m_robotAccelerationVariable.size);
+
+    return true;
+}
+
+bool SE3Task::update()
+{
+    m_b = -iDynTree::toEigen(m_kinDyn->getFrameBiasAcc(m_frameIndex));
+
+    m_SO3Controller.setState(
+        {BipedalLocomotion::Conversions::toManifRot(
+             m_kinDyn->getWorldTransform(m_frameIndex).getRotation()),
+         iDynTree::toEigen(m_kinDyn->getFrameVel(m_frameIndex).getAngularVec3())});
+
+    m_R3Controller.setState(
+        {iDynTree::toEigen(m_kinDyn->getWorldTransform(m_frameIndex).getPosition()),
+         iDynTree::toEigen(m_kinDyn->getFrameVel(m_frameIndex).getLinearVec3())});
+
+    m_SO3Controller.computeControlLaw();
+    m_R3Controller.computeControlLaw();
+
+    m_b.head<3>() += m_R3Controller.getControl().coeffs();
+    m_b.tail<3>() += m_SO3Controller.getControl().coeffs();
+
+    // Workaround because matrix view is not compatible with Eigen::Ref
+    // https://github.com/robotology/idyntree/issues/797
+    if (!m_kinDyn->getFrameFreeFloatingJacobian(m_frameIndex, m_jacobian))
+    {
+        std::cerr << "[SE3Task::update] Unable to get the jacobian." << std::endl;
+        return false;
+    }
+    this->subA(m_robotAccelerationVariable) = m_jacobian;
+
+    return true;
+}
+
+bool SE3Task::setReferenceTrajectory(const manif::SE3d& I_H_F,
+                                     const manif::SE3d::Tangent& mixedVelocity,
+                                     const manif::SE3d::Tangent& mixedAcceleration)
+{
+
+    bool ok = true;
+    ok = ok && m_R3Controller.setDesiredState({I_H_F.translation(), mixedVelocity.v()});
+    ok = ok && m_R3Controller.setFeedForward(mixedAcceleration.v());
+
+    ok = ok && m_SO3Controller.setDesiredState({I_H_F.quat(), mixedVelocity.w()});
+    ok = ok && m_SO3Controller.setFeedForward(mixedAcceleration.w());
```

Comment:
@GiulioRomualdi, Just to notify you of an upcoming change of these getter method names in https://github.com/artivis/manif/pull/190.